plugins/builds/index.js

```diff
@@ -761,9 +794,14 @@ const buildsPlugin = {
                         parentBuildId: current.build.id
                     };
 
-                    newBuild = await createInternalBuild(internalBuildConfig);
-                } else {
-                    // nextBuild is not build model, so fetch proper build
+                    try {
+                        newBuild = await createInternalBuild(internalBuildConfig);
+                    } catch (err) {
+                        nextBuild = await getNextBuild(nextBuildConfig);
```

Comment:
isn't this redundant because of line 778?

Reply:
so this is a try catch for race condition, build creation fail if there's already a build created for that job. Since we are not using infinite retry lock, so just adding this try catch as a fall back mechanism.